### PR TITLE
Removed iterative from walsh

### LINKED ISSUE #385: Remove iterative from walsh
Based on the discussions here
https://github.com/PTB-MR/mrpro/issues/340#issuecomment-2206273428

- [x] find out how many power iterations are needed for walsh -> 10 iterations are sufficient
- [ ] rename `iterative_walsh` to `walsh`
- [ ] remove `power_iterations` as an argument
- [ ] add n_power_iterations = X at the beginning 

for an example have a look at:
https://github.com/PTB-MR/mrpro/blob/faddf9b1460f582f2de36210f13eb92a5b665ca9/src/mrpro/algorithms/csm/inati.py#L33

### PR DESCRIPTION
**Pull request by bernharb**
Closes #385

### CODE REVIEW

File: tests/algorithms/csm/test_walsh.py [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-from mrpro.algorithms.csm import iterative_walsh
+from mrpro.algorithms.csm import walsh
 from mrpro.data import SpatialDimension
 from tests.algorithms.csm.conftest import multi_coil_image
 from tests.helper import relative_image_difference
 
 
-def test_iterative_Walsh(ellipse_phantom, random_kheader):
-    """Test the iterative Walsh method."""
+def test_Walsh(ellipse_phantom, random_kheader):
```
Comment (fzimmermann89) [inline tests/algorithms/csm/test_walsh.py]: while you are at it: i would not capitalize walsh here, as the function is also named "walsh" not "Walsh"

**Comment by fzimmermann89** [comment]
Do you use pre-commit?

If not yet, this is how to:
do a `pip install -e ".[test]"` in the  directory of your local clone of mrpro.
This will install "pre-commit", a tool that runs ruff for formatting, mypy for type checking etc.
Now you can call `pre-commit run --all-files` to check the files, let it apply some automatic formatting. Maybe you need to change some things manually, such that pre-commit does not raise any issues anymore. This would solve the "pre-commit" action failing.

**Comment by fzimmermann89** [comment]
If you use vscode, you can  use the "test" tab on the right left to see all tests and run them.
All of them, especially those related to your change, should pass.

Currently, there is an issue with the 
```
 from tests.algorithms.csm.test_iterative_walsh import multi_coil_image
E   ModuleNotFoundError: No module named 'tests.algorithms.csm.test_iterative_walsh'
```
as you forgot the change the name there :) 


ps: if you already know all these things, ignore my comments.
pps: if you did not know them and need more help -> slack.

**Review by fzimmermann89 — APPROVED** [review]
👍

**Comment by fzimmermann89** [comment]
Good job getting your first change into mrpro! 🎉 

One final tiny remark: Git commit messages are mostly in imperative mode. 
So, the final merged git commit should have been "Remove 'iterative' from walsh' -- not "Removed ... "
Not really important, just a small detail for the future :) 

@bernharb

**Comment by bernharb** [comment]
@fzimmermann89 

Alright, I'll remember it for future commits. I appreciate the feedback! :)

**Pull request merged**